spec/dynamoid/transaction_write/update_spec.rb

```diff
@@ -0,0 +1,192 @@
+      context 'simple primary key' do
+        before do
+          klass.create_table
+        end
+        it 'with attribute outside transaction' do
+          obj1 = klass.create!(name: "one")
+          obj1.name = "oneone"
+          Dynamoid::TransactionWrite.execute do |txn|
+            txn.update! obj1
+          end
+          obj1_found = klass.find(obj1.id)
+          expect(obj1_found).to eql(obj1)
+          expect(obj1_found.name).to eql("oneone")
+        end
+        it 'with attribute in transaction' do
+          obj2 = klass.create!(name: "two")
+          Dynamoid::TransactionWrite.execute do |txn|
+            txn.update! obj2,  {name: "twotwo"}
+          end
+          obj2_found = klass.find(obj2.id)
+          expect(obj2_found).to eql(obj2)
+          expect(obj2_found.name).to eql("twotwo")
+        end
+        it 'with class updates in transaction' do
+          obj3 = klass.create!(name: "three")
+          Dynamoid::TransactionWrite.execute do |txn|
+            txn.update! klass, {id: obj3.id, name: "threethree"}
+          end
+          obj3_found = klass.find(obj3.id)
+          expect(obj3_found).to eql(obj3)
+          expect(obj3_found.name).to eql("threethree")
+        end
+      end
+
+      context 'composite key' do
+        before do
+          klass_with_composite_key.create_table
+        end
+        it 'with attribute outside transaction' do
+          obj1 = klass_with_composite_key.create!(name: "one", age: 1)
+          obj1.name = "oneone"
+          Dynamoid::TransactionWrite.execute do |txn|
+            txn.update! obj1
+          end
+          obj1_found = klass_with_composite_key.find(obj1.id, range_key: 1)
+          expect(obj1_found).to eql(obj1)
+          expect(obj1_found.name).to eql("oneone")
+        end
+        it 'with attribute in transaction' do
+          obj2 = klass_with_composite_key.create!(name: "two", age: 2)
+
+          Dynamoid::TransactionWrite.execute do |txn|
+            txn.update! obj2, {name: "twotwo"}
+          end
+          obj2_found = klass_with_composite_key.find(obj2.id, range_key: 2)
+          expect(obj2_found).to eql(obj2)
+          expect(obj2_found.name).to eql("twotwo")
+        end
+        it 'with class updates in transaction' do
+          obj3 = klass_with_composite_key.create!(name: "three", age: 3)
+          Dynamoid::TransactionWrite.execute do |txn|
+            txn.update! klass_with_composite_key, {id: obj3.id, age: 3, name: "threethree"}
+          end
+          obj3_found = klass_with_composite_key.find(obj3.id, range_key: 3)
+          expect(obj3_found).to eql(obj3)
+          expect(obj3_found.name).to eql("threethree")
+        end
+      end
+
+      it 'updates timestamps of instance' do
+        klass.create_table
+        obj1 = klass.new(name: "one", created_at: Time.now - 48.hours, updated_at: Time.now - 24.hours)
+        Dynamoid::TransactionWrite.execute do |txn|
+          txn.create! obj1
```

Comment:
Is it supposed to be the `create!` method call?

Reply:
Yes. Fixed.